### PR TITLE
Monitor now has a status, status will be updated on new check, #235

### PR DESCRIPTION
**Pull request by ajhollid**
Monitor now has a status that is updated whenever a new check is created.

Logic in place for sending an alert if status changes from up to down or down to up.

### CODE REVIEW

File: Server/models/Check.js [hunk 1]
```diff
@@ -31,4 +31,27 @@ const CheckSchema = mongoose.Schema(
...
+    }
+
+    if (monitor.status === false && this.status === true) {
+      // TODO issue alert
+      console.log("Monitor went up");
+    }
+    await monitor.save();
+  }
+
+  next;
```
Comment (danielcj2) [inline Server/models/Check.js:54]: Is this a typo? Shouldn't it be `next()` instead?

File: Server/models/Check.js [hunk 2]
```diff
@@ -31,4 +31,27 @@ const CheckSchema = mongoose.Schema(
...
 );
 
+/**
+ * When a check is created, we should update the stauts of the associated monitor
+ * if it has changed.  Monitor starts with default status: true
+ */
+
+CheckSchema.pre("save", async function (next) {
+  const monitor = await mongoose.model("Monitor").findById(this.monitorId);
+  if (monitor.status !== this.status) {
```
Comment (MuhammadKhalilzadeh) [inline Server/models/Check.js:41]: please put this whole code block inside an if statement for error handling purposes, like this:

```
if (monitor) {
// other if(s)
} else {
// report that no monitors are found
}
```

File: Server/models/Check.js [hunk 3]
```diff
@@ -31,4 +31,27 @@ const CheckSchema = mongoose.Schema(
   }
 );
 
+/**
+ * When a check is created, we should update the stauts of the associated monitor
```
Comment (MuhammadKhalilzadeh) [inline Server/models/Check.js:35]: I'm trying to learn more and more about documentation these days and try to take advantage of AI and Chat-GPT tools for such purposes. So, here's what I got as another suggestion of comments and documentation for this code snippet:

```
/**
 * Middleware function executed before saving a check.
 * Updates the status of the associated monitor if it has changed.
 *
 * @param {function} next - Callback to proceed to the next middleware.
 */
```

What do you think?
Can we replace it with this one?

File: Server/models/Check.js (hunk 1 above)
Comment (MuhammadKhalilzadeh) [reply Server/models/Check.js:54]: Yeah. its a function:  `next()`

**Review by MuhammadKhalilzadeh — APPROVED** [review]
@ajhollid  You've done a cool job as always buddy. 👍😊
I hope my comments do help to improve ✌️

File: Server/models/Check.js (hunk 3 above)
Comment (ajhollid) [reply Server/models/Check.js:35]: AI can be a helpful tool, but I prefer to write my own documentation.  For example in this case it has missed out on the fact that the default value of the monitor status is true, which is useful information for anyone working on this method.

The js DOC is good though, we can include that :+1:

File: Server/models/Check.js (hunk 1 above)
Comment (ajhollid) [reply Server/models/Check.js]: Good catch, typing too fast :P

File: Server/models/Check.js (hunk 2 above)
Comment (ajhollid) [reply Server/models/Check.js]: Good call, I've added some error handling

File: Server/models/Check.js (hunk 3 above)
Comment (MuhammadKhalilzadeh) [reply Server/models/Check.js:35]: Yeah. We can combine them and improve it 👍🏻

**Pull request merged**